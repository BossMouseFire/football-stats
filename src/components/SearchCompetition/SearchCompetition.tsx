import React, { ChangeEvent } from 'react';
import { Input } from 'antd';
import { useAppDispatch } from '../../hooks/useAppSelector';
import competitionAction from '../../store/competition/competitionAction';

import styles from './SearchCompetiton.module.scss';

export const SearchCompetition = () => {
    const dispatch = useAppDispatch();

    return (
        <Input
            className={styles.search}
            placeholder='Введите название соревнования'
            onChange={handleSearchCompetition}
        />
    );

    function handleSearchCompetition(event: ChangeEvent<HTMLInputElement>) {
        const name = event.target.value;
        dispatch(competitionAction.setFilter(name));
    }
};
