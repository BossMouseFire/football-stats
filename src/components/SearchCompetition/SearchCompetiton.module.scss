.search {
  font-size: 18px;
  width: 350px;
}
