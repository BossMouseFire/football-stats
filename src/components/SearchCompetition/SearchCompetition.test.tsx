import { SearchCompetition } from './SearchCompetition';
import { renderToJson } from '../../test/helpers/renderToJson';

it('SearchCompetition renders correctly', () => {
    const search = renderToJson(<SearchCompetition />, true);

    expect(search).toMatchSnapshot();
});

export {};
