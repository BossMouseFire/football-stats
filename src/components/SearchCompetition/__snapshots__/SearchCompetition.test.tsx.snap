// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchCompetition renders correctly 1`] = `
<input
  className="ant-input css-dev-only-do-not-override-2i2tap search"
  disabled={false}
  onBlur={[Function]}
  onChange={[Function]}
  onCompositionEnd={[Function]}
  onCompositionStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  placeholder="Введите название соревнования"
  style={Object {}}
  type="text"
  value=""
/>
`;
